packages/engine/src/interaction/functions/equippableFunctions.ts

```diff
@@ -12,12 +13,8 @@ export const equipEntity = (
   equippedEntity: Entity,
   attachmentPoint: EquippableAttachmentPoint = EquippableAttachmentPoint.RIGHT_HAND
 ): void => {
-  if (
-    !hasComponent(equipperEntity, EquipperComponent) &&
-    hasComponent(equippedEntity, NetworkObjectComponent) &&
-    !hasComponent(equippedEntity, EquippedComponent)
-  ) {
-    addComponent(equipperEntity, EquipperComponent, { equippedEntity, data: {} })
+  if (!hasComponent(equipperEntity, EquipperComponent) && !hasComponent(equippedEntity, EquippedComponent)) {
```

Comment:
for my own understanding, is this `if` statement here to prevent the existing component data from being overwritten?

Reply:
Yes.